app/models/ability.rb

```diff
@@ -0,0 +1,39 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    return unless user.present?
+
+    can :read, User
+    can :read, Post
+    can :read, Comment
+    can :read, Like
+    can :create, Like
+    can :manage, Post, author_id: user.id
+    can :manage, Comment, author_id: user.id
+
+    return unless user.admin?
+
+    can :manage, Post
+    can :manage, Comment
```

Comment:
- [ ] In the case of a user being an admin please add permission to make it control(manage) all. because the admin supposes to have all permission.
[Hint] Because you add ` user.admin?` method inside the user model you can simply do this:
```ruby
class Ability
 include CanCan::Ability

     def initialize(user)
         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user 
         can :manage, Comment, author: user
         can :read, :all
         end
     end
 end
```

Reply:
Thank you, I've implemented these simplifications.